src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+       'tooltipFn': function (d) {
+         return '<span class="donut-tooltip-pf"style="white-space: nowrap;">' +
+                   d[0].value + ' ' + d[0].name +
+                '</span>';
+       },
+       'centerLabelFn': function (scope) {
+         return '<tspan dy="0" x="0" class="donut-title-big-pf">' +
+                  scope.data.available + '</tspan>' +
+                '<tspan dy="20" x="0" class="donut-title-small-pf">' +
+                  'Free' +
+                '</tspan>';
+       }
+     };
+
+     $scope.custData = {
+         'used': '670',
+         'total': '1000'
+     };
+
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin', '$timeout',
+  function(chartsMixin, $timeout) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        id: '@',
+        config: '=',
+        data: '=',
+        centerLabel: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/donut/donut-pct-chart.html',
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
+          };
+
+          $scope.statusDonutColor = function(scope) {
+            var color = {
+              pattern: []
+            };
+            var percentUsed = scope.data.used / scope.data.total * 100.0;
+            color.pattern[0] = $scope.getStatusColor(percentUsed, scope.config.thresholds);
+            color.pattern[1] = '#D1D1D1';
+            return color;
+          };
+
+          var donutTooltip = function(scope) {
+            return {
+              contents: function (d) {
+                if (scope.config.tooltipFn) {
+                  var tootipHtml = '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                                        scope.config.tooltipFn(d) +
+                                   '</span>';
+                  return tootipHtml;
+                } else {
+                  return '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                            Math.round(d[0].ratio * 100) + '%' + ' ' + $scope.config.units + ' ' + d[0].name +
+                         '</span>';
+                }
```

Comment:
Hi, you wrote: "Do we have a coding standard on var blah = function(variable){}; vs var blah = function (variable) {};? I see both in this PR."
Honestly, it took me a few re-reads to figure out the difference between A vs B (a single space!?).   Are we really going to be doing code reviews to this level of detail?  If so, code reviews are going to take much longer to do, and I'm not sure holding developers to such ridged standards is productive, or really makes the code any better.
I agree with many of the other best practice issues you have brought up, but I think we need to come to an agreement on what best practices it makes sense to follow.
Thanks,
- Dave